src/board/registers.ts

```diff
@@ -48,7 +66,102 @@ export class Registers {
    * @param word word to write into the register
    */
   public writeRegister(register: Register, word: Word): void {
-    this.data[register] = word.value
+    this.data[register] = word
+  }
+
+  /**
+   * checks wheter the desired flag is set or not
+   * @param flag Flag to read
+   * @returns true if flag is set
+   */
+  public isFlagSet(flag: Flag): boolean {
+    let bitOffset = this.getFlagOffsetWithinAPSR(flag)
+    return this.data[Register.APSR].isBitSet(bitOffset)
+  }
+
+  /**
+   * change bit value of specific flag
+   * @param flag flag to change
+   * @param setBit with true set to 1 otherwise 0
+   */
+  public setFlag(flag: Flag, setBit: boolean) {
+    let offset = this.getFlagOffsetWithinAPSR(flag)
+    this.data[Register.APSR] = setBit
+      ? this.data[Register.APSR].setBit(offset)
+      : this.data[Register.APSR].clearBit(offset)
+  }
+  /**
+   * Updates just the overflow flag
+   * @param value
+   */
+  public setNegativeFlag(value: Word) {
+    this.setFlag(Flag.N, value.hasSign())
+  }
+  /**
+   * Updates just the overflow flag
+   * @param value
+   */
+  public setZeroFlag(value: number) {
+    this.setFlag(Flag.Z, value === 0)
+  }
+  /**
+   * Updates just the carry flag
+   * @param isCarrySet
+   */
+  public setCarryFlag(isCarrySet: boolean) {
+    this.setFlag(Flag.C, isCarrySet)
+  }
+  /**
+   * Updates just the overflow flag
+   * @param isCarrySet
+   * @param wasCarryHighestBit
+   */
+  public setOverflowFlag(isCarrySet: boolean, wasCarryHighestBit: boolean) {
+    this.setFlag(Flag.V, wasCarryHighestBit ? !isCarrySet : isCarrySet)
+  }
+
+  /**
+   * Sets all flags in one go
+   * @param value resulting value for which flags should be set
+   * @param isCarrySet if carry happend during operation
+   * @param wasCarryHighestBit if there was a carry to highest bit (31)
+   */
+  public setFlags(
+    value: Word,
+    isCarrySet: boolean = false,
+    wasCarryHighestBit: boolean = false
```

Comment:
Couldn't we just use the highest bit of `value` instead of passing `wasCarryHighestBit`?

Reply:
Maybe it should be renamed to `didHighestBitChange`?

Reply:
What about the addtion if I add 2 negative values and the result is again a negative value. that means the highest bit did not change but there was a carry to the highest bit and the general carry flag should be set as well

Reply:
what about `wasCarryToHighestBit`?

Reply:
What about `wasCarryIntoSign`?

Reply:
suggestion taken. thx!